### DISCUSSION TITLE
What's required to consider JReleaser as ready for Bisq?

### DISCUSSION THREAD

**Discussion by ghubstan** [jreleaser/jreleaser#379] [General]
>> RE https://github.com/bisq-network/projects/issues/60#issuecomment-925899567

Mine is not the final say, but I am confident Bisq would start using JReleaser to package Bisq releases when

1. Your fix for https://github.com/jreleaser/jreleaser/issues/363  is released [milestone 0.7.0 item]

  And

2. [Support for assembling native packages with jpackage](https://github.com/jreleaser/jreleaser/issues/207) is released  [roadmap 1.0 item]

There is another useful [jpackage wrapper plugin](https://github.com/petr-panteleyev/jpackage-gradle-plugin), but my personal preference would be to use one or the other project for both the jdk download/verify and packaging tasks.

Thanks!

**Comment by aalmiray** [comment]
Thanks @ghubstan. Regarding jpackage support as laid out in #207, decided to let it rest for a while for the following reasons:

 1. JReleaser's main concern is to release, package, and announce binaries not create or assemble them, that's the job for the build tool.
 2. Despite the previous statement, JReleaser added support for building cross-platform Java Runtimes with Jlink as to my knowledge there's no easy way to do that with the current crop of Maven/Gradle plugins. Nothing stops developers from building custom JRs with other tools though, this is just a convenience.
 3. This also applies to GraalVM Native Images. Popular microservice frameworks (Spring Boot, Micronaut, Quarkus, Helidon) already offer ready-made configuration to build such binaries though they do not offer native executable distributions as required by JReleaser, but that's a matter of applying the appassembler/assembly plugins (Maven) or application/distribution plugins (Gradle).
 
In the case of Gradle I'd recommend using the [badass-jlink](https://badass-jlink-plugin.beryx.org/releases/latest/) plugin (modular app) or [badass-runtime](https://badass-runtime-plugin.beryx.org/releases/latest/) plugin (non-modular) app, as they make it very easy to create a custom JR or platform installer with jpackage. This is what I use for building platform packages for Ikonli (a JavaFX app)

https://github.com/kordamp/ikonli/blob/23eb61b0e67a692b4da9f58371a8611effd68fdb/apps/ikonli-browser/ikonli-browser.gradle

Which as you may see in the following JReleaser configuration file it's just a matter of defining artifact paths, the Gradle build takes care of the rest

https://github.com/kordamp/ikonli/blob/23eb61b0e67a692b4da9f58371a8611effd68fdb/jreleaser.yml

**Reply by ghubstan** [reply]
Thanks for setting me straight on the scope of JReleaser, your intent to let #207 rest awhile, and all of your advice, including the badass recommendation.  I'm busy with other tasks right now but do intend to investigate all of this.

And very kind of you to move my post from issues to discussions.  I know I had made a mistake when I hit the submit button and could not delete the new issue.  Oops.
